packages/hooks/src/useDebounceEffect/index.ts

```diff
@@ -0,0 +1,24 @@
+import { useEffect, EffectCallback, DependencyList, useState } from 'react';
+import { DebounceOptions } from '../useDebounce/debounceOptions';
+import useDebounceFn from '../useDebounceFn';
+import useUpdateEffect from '../useUpdateEffect';
+
+function useDebounceEffect(
+  effect: EffectCallback,
+  deps?: DependencyList,
+  options?: DebounceOptions,
+) {
+  const [flag, setFlag] = useState({});
+
+  const { run } = useDebounceFn(() => {
+    setFlag({});
+  }, options);
+
+  useEffect(() => {
+    return run();
+  }, deps);
+
+  useUpdateEffect(effect, [flag]);
+}
```

Comment:
deps 变化 -> run -> flag -> effectcallback

---
可不可以直接用 deps -> effectcallback 呢?

Reply:
哦!
是 debounceEffect 不是 debounceUpdateEffect

run -> flag 是为了在 第一次 useEffect 中也进行触发

---

但是这样的话为什么不直接这样:
```suggestion
function useDebounceEffect(
  effect: EffectCallback,
  deps?: DependencyList,
  options?: DebounceOptions,
) {
  const { run } = useDebounceFn(() => {
    effect({});
  }, options);

  useEffect(() => {
    return run();
  }, deps);
}
```